mesh_generation/generate_mesh.py

```diff
@@ -29,35 +31,46 @@
 import pandas as pd
 
 
-def is_git_repo():
+def get_git_url(file):
     """
-    Return True/False depending on whether or not the current directory is a git repo.
+    If the provided file is in a git repo, return the url to its most recent commit remote.origin.
     """
+    dirname = os.path.dirname(file)
 
-    return subprocess.call(
-        ['git', '-C', '.', 'status'],
-        stderr=subprocess.STDOUT,
-        stdout = open(os.devnull, 'w')
-    ) == 0
+    try:
+        url = subprocess.check_output(
+            ["git", "-C", dirname, "config", "--get", "remote.origin.url"]
+        ).decode("ascii").strip()
+        url = url.removesuffix('.git')
+    except subprocess.CalledProcessError:
+        return None
 
-def git_info():
-    """
-    Return the git repo origin url, relative path to this file, and latest commit hash.
-    """
+    if url.startswith("git@github.com:"):
+        url = f"https://github.com/{url.removeprefix('git@github.com:')}"
 
-    url = subprocess.check_output(
-        ["git", "remote", "get-url", "origin"]
-    ).decode('ascii').strip()
     top_level_dir = subprocess.check_output(
-        ['git', 'rev-parse', '--show-toplevel']
-    ).decode('ascii').strip()
-    rel_path = os.path.relpath(__file__, top_level_dir)
+        ["git", "-C", dirname, "rev-parse", "--show-toplevel"]
+    ).decode("ascii").strip()
+    rel_path = file.removeprefix(top_level_dir)
+
     hash = subprocess.check_output(
-        ['git', 'rev-parse', 'HEAD']
-    ).decode('ascii').strip()
+        ["git", "-C", dirname, "rev-parse", "HEAD"]
```

Comment:
could use `--short` if you'd rather a shorter git hash, though full hash is safer...